build-tools/src/main/java/org/elasticsearch/gradle/util/GradleUtils.java

```diff
@@ -49,23 +49,8 @@ public static SourceSetContainer getJavaSourceSets(Project project) {
         return project.getConvention().getPlugin(JavaPluginConvention.class).getSourceSets();
     }
 
-    public static <T extends Task> TaskProvider<T> maybeRegister(TaskContainer tasks, String name, Class<T> clazz, Action<T> action) {
-        try {
-            return tasks.named(name, clazz);
-        } catch (UnknownTaskException e) {
-            return tasks.register(name, clazz, action);
-        }
-    }
-
     public static void maybeConfigure(TaskContainer tasks, String name, Action<? super Task> config) {
-        TaskProvider<?> task;
-        try {
-            task = tasks.named(name);
-        } catch (UnknownTaskException e) {
-            return;
-        }
-
-        task.configure(config);
+        tasks.matching(t -> t.getName().equals(name)).configureEach( t-> config.execute(t));
```

Comment:
Calling `matching` here doesn't eagerly resolve all tasks? Wouldn't it have to if the predicate filtered on anything other than name?

Reply:
no it doesn't